Update UFS to Sept 9 version

WalterKolczynski-NOAA opened the pull request:
**Description**
Updates the UFS version. This captures the conversion of UFS module from TCL to lua. A couple of the CICE namelist variables are no longer valid in this version, so they are removed.

Due to memory limitations on Hera and the increased memory requirements of GOCART, the number of threads is increased there when running the forecast with aerosols.

Also added a temporary block to delete any existing gocart output files. The ability to clobber files was deactivated a while ago and I got fed up with forecast jobs failing on retry.

**Type of change**
- [x] Maintenance

**How Has This Been Tested?**
- [x] Cycled test on Orion
- [x] Coupled test on Orion
- [x] ATMA test on Hera
- [x] Coupled test on Hera
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

WalterKolczynski-NOAA committed:
Remove existing aerosol output

Due to a change, GOCART currently will not clobber existing output files,
resulting in a model crash when attempting to rerun. So, we temporarily
remove the aerosol output until an option is introduced to allow clobber.

WalterKolczynski-NOAA committed:
Update UFS to Sept 9

Updates the UFS model to Sept 9. Some modules are updated to match the UFS
versions.

This update is still not fully tested and additional changes may be necessary.

WalterKolczynski-NOAA committed:
Adjust fcst resources on Hera for aerosols

Hera has less memory per core than other machines. When UFS is run
with GOCART, this results in crashes due to the memory limit being
reached. To avoid this, the number of threads is increased on Hera
when running with aerosols.

WalterKolczynski-NOAA committed:
Style fixes

JessicaMeixner-NOAA commented on modulefiles/modulefile.ww3.orion.lua:
```diff
@@ -23,3 +23,4 @@ load(pathJoin("hdf5", "1.10.6"))
 load(pathJoin("netcdf", "4.7.4"))
 
 load(pathJoin("w3nco", "2.4.1"))
```
I think you can also remove w3nco.  Do you have paths to the output of your coupled model run by any chance I could look at?

WalterKolczynski-NOAA replied:
Pretty sure it wouldn't build without it, which is why I added it. It was probably a month ago at this point, so memory is fuzzy.

`/scratch2/NCEPDEV/ensemble/noscrub/Walter.Kolczynski/global-workflow/update_ufs_cpl`

JessicaMeixner-NOAA replied:
We don't run with it in the standalone WW3 regression tests, which is why I'm pretty sure you can remove it.  Maybe I can run a quick test to see.  It's not a big deal to carry around the extra module but this module list has been trimmed down to not have any extras, which is why I pointed it out.  Thanks for the path!

aerorahul replied:
I think so too.
`w3nco` is no longer required in any library/executable, and if it is, we need to snuff it out.
I know it is being automatically loaded when `nemsio/2.5.2` is loaded.  It was removed from `nemsio/2.5.3+`

If necessary, we can hold this change for a follow up PR since testing for this PR has completed.

WalterKolczynski-NOAA replied:
Sorry, my wires were crossed. I was thinking of the thing I added, but this is about removing the thing it basically replaced. Doing a test build just to be sure, but should work.

WalterKolczynski-NOAA replied:
Done. Also changed the modules on wcoss2 and s4 over to w3emc, as those machines weren't in the repo when I first made these changes.

aerorahul commented on ush/parsing_namelists_CICE.sh:
```diff
@@ -80,8 +80,6 @@ cat > ice_in <<eof
    restart_FY   = .false.
    tr_lvl       = .true.
    restart_lvl  = .false.
-   tr_pond_cesm = .false.
```
can you add a comment in the PR description, so this does not seem like trying to sneak in?

WalterKolczynski-NOAA replied:
Done

aerorahul reviewed (approved):
A couple of comments.
As long as the other reviewers are satisfied, this PR can be merged.

WalterKolczynski-NOAA committed:
Remove unneeded w3nco module load

With the move to a w3emc that includes w3nco, the latter no longer
needs to be loaded to build WW3 pre/post.

Additionally, added the w3emc module loads for the two machines that
were not supported when this update process started.

JessicaMeixner-NOAA reviewed (approved):
The updates match what I see here: https://github.com/ufs-community/ufs-weather-model/compare/Prototype-P8...3c3548d

Thanks @WalterKolczynski-NOAA !
